cmd/restic/cmd_self_update.go

```diff
@@ -56,11 +57,18 @@ func runSelfUpdate(opts SelfUpdateOptions, gopts GlobalOptions, args []string) e
 
 	fi, err := os.Lstat(opts.Output)
 	if err != nil {
-		return err
-	}
-
-	if !fi.Mode().IsRegular() {
-		return errors.Errorf("output file %v is not a normal file, use --output to specify a different file", opts.Output)
+		dirname := filepath.Dir(opts.Output)
+		di, err := os.Lstat(dirname)
+		if err != nil {
+			return err
+		}
+		if !di.Mode().IsDir() {
+			return errors.Errorf("output parent path %v is not a directory, use --output to specify a different file path", dirname)
```

Comment:
Please use `errors.Fatalf` to create the error, otherwise the error looks like restic just exploded ;-)

Reply:
What about the fact that the corresponding error message I'm replacing uses `errors.Errorf`?

Reply:
An error message should not look like this:
```
output path tmp is not a normal file, use --output to specify a different file path
main.runSelfUpdate
	restic/restic/cmd/restic/cmd_self_update.go:70
main.glob..func23
	restic/restic/cmd/restic/cmd_self_update.go:30
github.com/spf13/cobra.(*Command).execute
	go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:842
github.com/spf13/cobra.(*Command).ExecuteC
	go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:950
github.com/spf13/cobra.(*Command).Execute
	go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:887
main.main
	restic/restic/cmd/restic/main.go:86
runtime.main
	go/1.15.2/libexec/src/runtime/proc.go:204
runtime.goexit
	go/1.15.2/libexec/src/runtime/asm_amd64.s:1374
```
The following is much nicer:
`Fatal: output path tmp is not a normal file, use --output to specify a different file path`
So the current behavior needs fixing. If you take a look at line 78, there `errors.Fatalf` is already used.

Reply:
Ack.

I have now (force) pushed a new commit, replacing `errors.Errorf` with `errors.Fatalf`.